view/frontend/web/template/payment/amwal-payment/form.html

```diff
@@ -28,27 +29,10 @@
         </div>
         <div class="actions-toolbar">
             <div class="primary">
-                <div data-bind="
-                attr: {'id': getCode() + '-button'}">
-                    <amwal-checkout-button
-                        id="amwal-place-order-button"
-                        data-bind="attr: {
-                            'ref-id': getRefId(),
-                            'merchant-id': getMerchantId(),
-                            'country-code': getCountryCode(),
-                            'locale': getLocale(),
-                            'dark-mode': getDarkMode(),
-                            'test-environment': getTestEnvironment(),
-                            'allowed-address-countries': getAllowedCountries(),
-                            'enable-pre-checkout-trigger': true,
-                            'enable-pre-pay-trigger': true,
-                        }"
-                        amount="0"
-                        address-required="false">
-                    </amwal-checkout-button>
+                <div id="amwal-place-order-button" trigger-context="regular-checkout" address-required="false" data-bind="attr: {}">
```

Comment:
`address_required` needs to be returned from the `/rest/V1/amwal/button/cart` endpoint to be initialized correctly in the button. `address_required` should only be false if the `trigger_context` is `regular-checkout`

```suggestion
                <div id="amwal-place-order-button" trigger-context="regular-checkout">
```